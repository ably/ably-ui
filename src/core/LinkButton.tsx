import React from "react";
import {
  ButtonPropsBase,
  commonButtonInterior,
  commonButtonProps,
} from "./Button";
import cn from "./utils/cn";

export type LinkButtonProps = ButtonPropsBase & {
  disabled?: boolean;
  onClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
} & React.AnchorHTMLAttributes<HTMLAnchorElement>;

const LinkButton: React.FC<LinkButtonProps> = ({
  variant = "primary",
  size,
  leftIcon,
  rightIcon,
  children,
  className,
  disabled,
  onClick,
  ...rest
}) => {
  const handleClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
    if (disabled) {
      e.preventDefault();
      return;
    }
    onClick?.(e);
  };

  return (
    <a
      {...commonButtonProps({
        variant,
        size,
        leftIcon,
        rightIcon,
        className: cn(className, {
          "ui-button-disabled dark:ui-button-disabled-dark": disabled,
        }),
      })}
      role="button"
      aria-disabled={disabled}
      onClick={handleClick}
      {...(rest as React.AnchorHTMLAttributes<HTMLAnchorElement>)}
    >
      {commonButtonInterior({ leftIcon, rightIcon, children })}
    </a>
  );
};

export default LinkButton;
