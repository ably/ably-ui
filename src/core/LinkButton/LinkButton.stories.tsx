import React from "react";
import { Meta, StoryFn } from "@storybook/react-vite";
import LinkButton, { LinkButtonProps } from "../LinkButton";

export default {
  title: "Components/Link Button",
  component: LinkButton,
  parameters: {
    docs: {
      description: {
        component:
          "A variant of the `Button` component that renders an `a` element instead of a `button` element, with all the typing constraints and props one would expect from an anchor.",
      },
    },
  },
} as Meta;

const Template: StoryFn<LinkButtonProps> = (args) => <LinkButton {...args} />;

export const Default = Template.bind({});
Default.args = {
  href: "#",
  variant: "primary",
  children: "Default Link Button",
};

export const External = Template.bind({});
External.args = {
  href: "https://www.ably.com",
  children: "External Link Button",
  target: "_blank",
  rel: "noopener noreferrer",
};

export const Disabled = Template.bind({});
Disabled.args = {
  href: "#",
  children: "Disabled Link Button",
  disabled: true,
};
