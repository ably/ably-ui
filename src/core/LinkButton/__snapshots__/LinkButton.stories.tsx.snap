// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Link Button Default smoke-test 1`] = `
<a class="ui-button-primary"
   role="button"
   href="#"
>
  Default Link Button
</a>
`;

exports[`Components/Link Button Disabled smoke-test 1`] = `
<a class="ui-button-primary ui-button-disabled dark:ui-button-disabled-dark"
   role="button"
   aria-disabled="true"
   href="#"
>
  Disabled Link Button
</a>
`;

exports[`Components/Link Button External smoke-test 1`] = `
<a class="ui-button-primary"
   role="button"
   href="https://www.ably.com"
   target="_blank"
   rel="noopener noreferrer"
>
  External Link Button
</a>
`;
